src/components/experiment-tracking/run-dataset/run-dataset.js

```diff
@@ -1,15 +1,40 @@
 import React from 'react';
+import ContentLoader from 'react-content-loader';
 import classnames from 'classnames';
 import Accordion from '../accordion';
 import PinArrowIcon from '../../icons/pin-arrow';
 import PlotlyChart from '../../plotly-chart';
 import { sanitizeValue } from '../../../utils/experiment-tracking-utils';
 import { TransitionGroup, CSSTransition } from 'react-transition-group';
+import { experimentTrackingLazyLoadingColours } from '../../../config';
 
 import getShortType from '../../../utils/short-type';
 import './run-dataset.css';
 import '../run-metadata/animation.css';
 
+const Loader = ({ x, y, length, theme }) => {
+  return (
+    <ContentLoader
+      viewBox="0 10 150 25"
+      width="200px"
+      height="100%"
+      backgroundColor={
+        theme === 'dark'
+          ? experimentTrackingLazyLoadingColours.backgroundDarkTheme
+          : experimentTrackingLazyLoadingColours.backgroundLightTheme
+      }
+      foregroundColor={
+        theme === 'dark'
+          ? experimentTrackingLazyLoadingColours.foregroundDarkTheme
+          : experimentTrackingLazyLoadingColours.foregroundLightTheme
+      }
+      speed={2}
+    >
+      <rect width="150" height="16" x={x} y={y + length * 2} />
+    </ContentLoader>
+  );
+};
```

Comment:
What's the thinking behind composing this component in this file? To me, it would be more natural if this lived in your `/run-dataset/run-dataset-loader.js` file. Then you'd just have one new import in this file and all the new loader composition is kept separate.

Reply:
originally i was thinking it should live inside the parent component as it only uses there but what you suggested also makes sense. have moved to the loader component